aggregate.go

```diff
@@ -77,11 +77,13 @@ type AggregateRoot interface {
 	// ApplyEvent updates the aggregate instance to reflect the occurrence of an
 	// event that was recorded against this instance.
 	//
-	// It MUST NOT be called with a message of any type that has not been
-	// configured for production by a prior call to Configure().
+	// The engine MUST call ApplyEvent() for each newly recorded event. It MAY
+	// call ApplyEvent() with any event that has already been recorded against
+	// this instance, even if that event type is no longer configured for
+	// production by a prior call to Configure().
 	//
-	// It MUST accept all messages of the types that have been configured for
-	// production, though any given call MAY be a no-op.
+	// The implementation MUST panic with the UnexpectedMessage value if called
+	// with any other event type. Any given call MAY be a no-op.
```

Comment:
This seem s a little bit confusing. What other "event type" is there?

Reply:
It's referring to event types other than those in the paragraph above. How about this?

```go
	// ApplyEvent updates the aggregate instance to reflect the occurrence of an
	// event that was recorded against this instance.
	//
	// The engine MUST call ApplyEvent() for each newly recorded event. It MAY
	// call ApplyEvent() with any event that has already been recorded against
	// this instance, even if that event type is no longer configured for
	// production by a prior call to Configure().
	//
	// The implementation MUST accept the event types as described above, though
	// any such call MAY be a no-op.
	//
	// The implementation MUST panic with the UnexpectedMessage value if called
	// with any event type other than those described above.
	ApplyEvent(m Message)
```

Reply:
Updated to above, Kev approved via Slack.